InterfaceFortran/fortester.f90

```diff
@@ -27,184 +27,1471 @@
+points  => tsgGetPoints(gridID);  i_b = tsgGetNumPoints(gridID)
+allocate( pointsb(2,i_a), double_2d_a(1,i_b), double_2d_b(1,i_a), double_2d_c(1,i_a) )
+rnd     => random(2,i_a)
+pointsb =  -1.d0 + 2.d0 * rnd
+double_2d_a(1,:) = exp( -points(1,:)**2 - 2.d0 * points(2,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+call tsgEvaluateBatch(gridID, pointsb, i_a, double_2d_b)
+do i = 1,i_a
+  double_pnt_1d_a  => tsgGetInterpolationWeights(gridID_II, pointsb(:,i))
+  double_2d_c(:,i) =  matmul(double_2d_a,double_pnt_1d_a)
+  deallocate(double_pnt_1d_a)
+enddo
+if ( norm2d(double_2d_b - double_2d_c) > 1.d-11 ) then
+  write(*,*) "Mismatch in tsgGetInterpolationWeights: global case"
+  stop 1
+endif
+deallocate(points, pointsb, double_2d_a, double_2d_b, double_2d_c, rnd)
+
+
+
+
+
+
+
+
+!=======================================================================
+!       tsgEstimateAnisotropicCoefficients()
+!=======================================================================
+call tsgMakeGlobalGrid(gridID, 2, 1, 9, tsg_level, tsg_rleja)
+points => tsgGetPoints(gridID);  i_b = tsgGetNumPoints(gridID)
+allocate( double_2d_a(1,i_b) )
+double_2d_a(1,:) = exp( points(1,:) + points(2,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+int_pnt_1d_a => tsgEstimateAnisotropicCoefficients(gridID,tsg_iptotal,1)
+if ( abs( int_pnt_1d_a(1)/dble(int_pnt_1d_a(2)) - 2.0 ) > 0.2 ) then
+  write(*,*) "Mismatch in tsgEstimateAnisotropicCoefficients: total degree"
+  stop 1
+endif
+deallocate(int_pnt_1d_a)
+int_pnt_1d_a => tsgEstimateAnisotropicCoefficients(gridID,tsg_ipcurved,1)
+if ( size(int_pnt_1d_a) .ne. 4 ) then
+  write(*,*) "Mismatch in tsgEstimateAnisotropicCoefficients: curved dimensions"
+  stop 1
+endif
+if ( (abs( int_pnt_1d_a(1)/dble(int_pnt_1d_a(2)) - 2.0 ) > 0.2) &
+     .or. (int_pnt_1d_a(3)>0) .or. (int_pnt_1d_a(4)>0) ) then
+  write(*,*) "Mismatch in tsgEstimateAnisotropicCoefficients: curved"
+  stop 1
+endif
+deallocate(int_pnt_1d_a)
+deallocate(points, double_2d_a)
+
+
+
+
+
+
+
+
+!=======================================================================
+!       tsgSetAnisotropicRefinement()
+!=======================================================================
+call tsgMakeSequenceGrid(gridID, 3, 1, 3, tsg_level, tsg_leja, levelLimits=(/3,2,1/))
+points => tsgGetPoints(gridID);  i_b = tsgGetNumPoints(gridID)
+if ( check_points( points(1,:), (/0.d0,-1.d0,1.d0,1.d0/sqrt(3.d0)/), (/12345.d0/) ) .or. &
+     check_points( points(2,:), (/0.d0,-1.d0,1.d0/), (/1.d0/sqrt(3.d0)/) ) .or. &
+     check_points( points(3,:), (/0.d0,1.d0/), (/-1.d0,1.d0/sqrt(3.d0)/) ) ) then
+  write(*,*) "Mismatch in tsgSetAnisotropicRefinement: limits in make"
+  stop 1
+endif
+allocate( double_2d_a(1,i_b) )
+double_2d_a(1,:) = exp( -points(1,:)**2 - points(2,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+call tsgSetAnisotropicRefinement( gridID, tsg_iptotal, 30, 1 )
+pointsb => tsgGetNeededPoints(gridID)
```

Comment:
* must delete `pointsb` after the checks and before you call `pointsb =>` again in line 1230.